src/doom/m_menu.c

```diff
@@ -1449,6 +1449,18 @@ boolean M_Responder (event_t* ev)
             key = key_menu_abort;
             joywait = I_GetTime() + 5;
         }
+
+        // Fill-in the savegame name if user press Fire on the joystick
+        if (saveStringEnter && joybmenu >= 0 && ev->data1&1)
+        {
+            // Create a savegame string
+            char savestring[SAVESTRINGSIZE];
+
+            memset(savestring, 0, SAVESTRINGSIZE);
+            M_snprintf(savestring, SAVESTRINGSIZE - 1, "JOYSTICK SLOT %i", saveSlot);
+            saveCharIndex = strlen(savestring);
+            memcpy(savegamestrings[saveSlot], savestring, SAVESTRINGSIZE);
+        }
```

Comment:
@fabiangreffrath I've been looking at the code. If `saveStringEnter` is True M_Responder() always returns True.

I think I know what's going on: using debug `printfs()` I discovered that the keyboard and the mouse only produce events when a key is pressed or released. The exception to this is if a key is kept pressed: 1) when the key is initially pressed a KEYPRESS event is generated. 2) The key continues pressed. For a time (about a second) no events are generated. 3) After this time has passed a KEYPRESS event is generated for as long as the key is pressed.

The joystick does not work like that. Joystick produces an event *always*, even when there is no joystick button pressed and joysticks are centered. I do not know if this was true in Vanilla or not.

The think is, in the tick the savestring is generated by pressing Fire in the joystick M_Responder() actually returns True and the event is consumed. However, the next tick a new joystick event is generated and this event has the Fire button pressed. It is this second tick after saving the game the ones that fires the weapon.

Solving this will require some redesigning of the joysticks interface. The status of the joystick can be save on a struct. When the status of the joystick is checked (at the beginning of the tick) an event is produced if a change is detected. This will make the joystick behave like a keyboard/mouse. Also, if I recall correctly this is how the joystick works on idtech-4 in RBDoom3-BFG. This scheme also allows to do tricks like use trigger axis as buttons: with the current implementation the Left and Right triggers in XBox-like gamepads are totally useless in Chocolate Doom.

Anyway... I think this will require further discussion in their own Issue.